src/world.rs

```diff
@@ -0,0 +1,214 @@
+use std::{
+    fs, io::Read, path::{Path, PathBuf}
+};
+
+use regex::Regex;
+use serde::Deserialize;
+
+use crate::{Error, Map, ResourceCache, ResourceReader};
+
+/// A World is a collection of maps and their layout in the game world.
+#[derive(Deserialize, PartialEq, Clone, Debug)]
+pub struct World {
+    /// The path first used in a ['ResourceReader'] to load this world.
+    #[serde(skip_deserializing)]
+    pub source: PathBuf,
+    /// The maps present in this world.
+    pub maps: Option<Vec<WorldMap>>,
+    /// Optional regex pattern to load maps.
+    patterns: Option<Vec<WorldPattern>>,
```

Comment:
I can understand why you have decided to make this member private, however as-is it doesn't really provide any meaningful information nor can it be used anywhere (I think). Maybe it's a good idea to just make them public, so that users can do stuff such as hot-reloading the pattern. Either that, or we get entirely rid of the patterns member. @bjorn What do you think?

Reply:
I think it would make sense for this data to be public, especially if we just provide some convenience functions on `World` and `WorldPattern` to check whether a file name matches it.

Reply:
Patterns has been made public.